compiler/util/files.cpp

```diff
@@ -434,12 +434,14 @@ void addSourceFiles(int numNewFilenames, const char* filename[]) {
     // WE SHOULDN"T TRY TO OPEN .h files, just .c and .chpl and .o
     if (!isCHeader(filename[i])) {
       FILE* testfile = openInputFile(filename[i]);
-      if (fscanf(testfile, "%c", &achar) != 1) {
-        USR_FATAL("source file '%s' is either empty or a directory",
-                  filename[i]);
-      }
+      if(testfile) {
+        if (fscanf(testfile, "%c", &achar) != 1) {
```

Comment:
If the filename does not exist, `testfile` will be `NULL`, and `fscanf` will segFault.

I verified that with this `NULL` guard, if the filename does not exist, a later part of the compiler will say that the filename does not exist.

Reply:
This is fine, but what command would you run to see this error? I see an expected error with a simple test:

```
$ chpl does-not-exist.chpl
error: opening does-not-exist.chpl: No such file or directory
```

Reply:
> This is fine, but what command would you run to see this error? I see an expected error with a simple test:
> 
> ```
> $ chpl does-not-exist.chpl
> error: opening does-not-exist.chpl: No such file or directory
> ```
```
chpl --ignore-errors-for-pass does-not-exist.chpl
```
Currently it segFaults. With this guard, it produces this error:
```
error: opening does-not-exist.chpl: No such file or directory
error: opening does-not-exist.chpl: No such file or directory
ParseFile: Unable to open "does-not-exist.chpl" for reading
```